packages/frontend-2/components/settings/workspaces/security/sso/Form.vue

```diff
@@ -0,0 +1,142 @@
+        name="providerName"
+        color="foundation"
+        show-label
+        label-position="left"
+        :rules="[isRequired, isStringOfLength({ minLength: 5 })]"
+        type="text"
+      />
+      <hr class="border-outline-3" />
+      <FormTextInput
+        v-model="formData.clientId"
+        :disabled="!!initialData?.clientId"
+        label="Client ID"
+        name="clientId"
+        color="foundation"
+        show-label
+        label-position="left"
+        :rules="[isRequired, isStringOfLength({ minLength: 5 })]"
+        type="text"
+      />
+      <hr class="border-outline-3" />
+      <FormTextInput
+        v-model="formData.clientSecret"
+        :disabled="!!initialData?.clientId"
+        label="Client secret"
+        name="clientSecret"
+        color="foundation"
+        show-label
+        label-position="left"
+        type="text"
+        :rules="[isRequired, isStringOfLength({ minLength: 5 })]"
+      />
+      <hr class="border-outline-3" />
+      <FormTextInput
+        v-model="formData.issuerUrl"
+        :disabled="!!initialData?.clientId"
+        label="Issuer URL"
+        name="issuerUrl"
+        color="foundation"
+        show-label
+        label-position="left"
+        type="text"
+        :rules="[isRequired, isUrl, isStringOfLength({ minLength: 5 })]"
+      />
+      <div class="flex gap-2 mt-4">
+        <FormButton
+          v-if="!initialData"
+          :disabled="!challenge"
+          color="primary"
+          type="submit"
+        >
+          Add
+        </FormButton>
+        <FormButton color="outline" @click="$emit('cancel')">Cancel</FormButton>
+      </div>
+    </div>
+  </form>
+</template>
+
+<script setup lang="ts">
+import { useForm } from 'vee-validate'
+import { isRequired, isStringOfLength, isUrl } from '~~/lib/common/helpers/validation'
+import { useLoginOrRegisterUtils } from '~/lib/auth/composables/auth'
+import { usePostAuthRedirect } from '~/lib/auth/composables/postAuthRedirect'
+import type { SsoFormValues } from '~/lib/workspaces/helpers/types'
+import { useMixpanel } from '~/lib/core/composables/mp'
+
+const props = defineProps<{
+  initialData?: SsoFormValues
+  workspaceSlug: string
+}>()
+
+defineEmits<{
+  (e: 'cancel'): void
+  (e: 'submit', data: SsoFormValues): void
+}>()
+
+const logger = useLogger()
+const apiOrigin = useApiOrigin()
+const postAuthRedirect = usePostAuthRedirect()
+const { challenge } = useLoginOrRegisterUtils()
+const mixpanel = useMixpanel()
+
+const formData = ref<SsoFormValues>({
+  providerName: props.initialData?.providerName ?? '',
+  clientId: props.initialData?.clientId ?? '',
+  clientSecret: props.initialData?.clientSecret ?? '',
+  issuerUrl: props.initialData?.issuerUrl ?? ''
+})
+
+const { handleSubmit } = useForm<SsoFormValues>()
+
+const handleCreate = () => {
+  const baseUrl = `${apiOrigin}/api/v1/workspaces/${props.workspaceSlug}/sso/oidc/validate`
```

Comment:
Why not use `new URL()` with its utils for setting query params (URL.searchParams), instead of doing all of this string concatenation?